pydantic_extra_types/mongo_objectId.py

```diff
@@ -0,0 +1,34 @@
+from typing import Any
+
+from bson import ObjectId
+from pydantic_core import core_schema
+from pydantic_core import PydanticCustomError
+
+
+class ObjectIdField(str):
+    @classmethod
+    def __get_pydantic_core_schema__(
+        cls, _source_type: Any, _handler: Any
+    ) -> core_schema.CoreSchema:
+        object_id_schema = core_schema.chain_schema(
+            [
+                core_schema.str_schema(),
+                core_schema.no_info_plain_validator_function(cls.validate),
+            ]
+        )
+        return core_schema.json_or_python_schema(
+            json_schema=object_id_schema,
+            python_schema=core_schema.union_schema(
+                [core_schema.is_instance_schema(ObjectId), object_id_schema]
+            ),
+            serialization=core_schema.plain_serializer_function_ser_schema(
+                lambda x: str(x)
+            ),
+        )
+
+    @classmethod
+    def validate(cls, value):
+        if not ObjectId.is_valid(value):
+            raise PydanticCustomError('invalid object id', 'invalid format')
+
+        return ObjectId(value)
```

Comment:
```suggestion
        if not ObjectId.is_valid(value):
            raise PydanticCustomError('invalid object id', 'invalid format')

        return ObjectId(value)
```
```suggestion
        try:
            return ObjectId(value)
        except bson.errors.InvalidId as invalid_id:
            raise PydanticCustomError('value_error', 'invalid format for MongoDB object identifier') from invalid_id
```

Reply:
this should avoid checking twice for if the value is valid and also adds the ObjectId error message to the traceback which is useful because it specifies the structure it must follow, and looks something like: 
`is not a valid ObjectId, it must be a 12-byte input or a 24-character hex string`